### PR TITLE
Get CoinbaseSmartContractFactory and ERC1271 to 100% coverage 

### PR DESCRIPTION
**Pull request by stevieraykatz**
As part of an ongoing effort to achieve 100% coverage for all contracts in this repo...

## CoinbaseSmartContractFactory
1. Added a missing test for `initCodeHash()` 
2. Added two new tests to catch missing branch in `createAccount` for cases when the account has already been intitialized and another case wherein the account cannot be initialized because of malformed owner bytes.

## ERC1271
Currently, foundry coverage doesn't like the fact that we return the return-declared `salt` and `extensions`. Adding an explicit check that the return values match the uninitialized args doesn't tick the coverage box. However, in uncommitted changes (shown below), explicitly defining the values of `salt` and `extensions` gets this test to pass with "full coverage". Since explicitly declaring these variables (again) adds ~50 gas, we shouldn't make the change just for the sake of the coverage report. 

```solidity
    function eip712Domain()
        external
        view
        virtual
        returns (
            bytes1 fields,
            string memory name,
            string memory version,
            uint256 chainId,
            address verifyingContract,
            bytes32 salt,
            uint256[] memory extensions
        )
    {
        fields = hex"0f"; // `0b1111`.
        (name, version) = _domainNameAndVersion();
        chainId = block.chainid;
        verifyingContract = address(this);
        // salt = salt; // `bytes32(0)`.
        // extensions = extensions; // `new uint256[](0)`.
        salt = bytes32(0);
        extensions = new uint256[](0); 
    }
```
 
Coverage report as of this PR:
```
| File                                             | % Lines          | % Statements     | % Branches      | % Funcs         |
|--------------------------------------------------|------------------|------------------|-----------------|-----------------|
| script/DeployERC4337Factory.s.sol                | 0.00% (0/8)      | 0.00% (0/12)     | 100.00% (0/0)   | 0.00% (0/2)     |
| src/CoinbaseSmartWallet.sol                      | 100.00% (46/46)  | 100.00% (65/65)  | 100.00% (22/22) | 100.00% (13/13) |
| src/CoinbaseSmartWalletFactory.sol               | 100.00% (10/10)  | 100.00% (10/10)  | 100.00% (4/4)   | 100.00% (4/4)   |
| src/ERC1271.sol                                  | 85.71% (12/14)   | 89.47% (17/19)   | 100.00% (2/2)   | 100.00% (6/6)   |
| src/MultiOwnable.sol                             | 100.00% (27/27)  | 100.00% (38/38)  | 80.00% (8/10)   | 100.00% (13/13) |
| src/utils/ERC1271InputGenerator.sol              | 0.00% (0/9)      | 0.00% (0/14)     | 0.00% (0/6)     | 0.00% (0/1)     |
| test/CoinbaseSmartWallet/SmartWalletTestBase.sol | 0.00% (0/16)     | 0.00% (0/18)     | 0.00% (0/1)     | 0.00% (0/6)     |
| test/CoinbaseSmartWallet/UpgradeToAndCall.t.sol  | 100.00% (1/1)    | 100.00% (1/1)    | 100.00% (0/0)   | 50.00% (1/2)    |
| test/MultiOwnable/MultiOwnableTestBase.t.sol     | 0.00% (0/3)      | 0.00% (0/3)      | 100.00% (0/0)   | 0.00% (0/1)     |
| test/mocks/MockCoinbaseSmartWallet.sol           | 50.00% (1/2)     | 66.67% (2/3)     | 100.00% (0/0)   | 50.00% (1/2)    |
| test/mocks/MockEntryPoint.sol                    | 40.00% (2/5)     | 33.33% (2/6)     | 0.00% (0/2)     | 33.33% (1/3)    |
| test/mocks/MockMultiOwnable.sol                  | 100.00% (1/1)    | 100.00% (1/1)    | 100.00% (0/0)   | 100.00% (1/1)   |
| test/mocks/MockTarget.sol                        | 80.00% (4/5)     | 100.00% (4/4)    | 0.00% (0/1)     | 66.67% (2/3)    |
| Total                                            | 70.75% (104/147) | 72.16% (140/194) | 75.00% (36/48)  | 73.68% (42/57)  |
```

### CODE REVIEW

File: test/CoinbaseSmartWalletFactory.t.sol [hunk 1]
```diff
@@ -32,6 +33,20 @@ contract CoinbaseSmartWalletFactoryTest is Test {
         factory.createAccount{value: 1e18}(owners, 0);
     }
 
+    function test_exitIfAccountIsAlreadyInitialized() public {
+        CoinbaseSmartWallet a = factory.createAccount{value: 1e18}(owners, 0);
```
Comment (wilsoncusack) [inline test/CoinbaseSmartWalletFactory.t.sol:37]: nit, don't need value 
```suggestion
        CoinbaseSmartWallet a = factory.createAccount(owners, 0);
```

File: test/CoinbaseSmartWalletFactory.t.sol [hunk 2]
```diff
@@ -32,6 +33,20 @@ contract CoinbaseSmartWalletFactoryTest is Test {
         factory.createAccount{value: 1e18}(owners, 0);
     }
 
+    function test_exitIfAccountIsAlreadyInitialized() public {
```
Comment (wilsoncusack) [inline test/CoinbaseSmartWalletFactory.t.sol:36]: I'm not sure we really test this here 🤔

File: test/CoinbaseSmartWalletFactory.t.sol [hunk 3]
```diff
@@ -32,6 +33,20 @@ contract CoinbaseSmartWalletFactoryTest is Test {
...
+        CoinbaseSmartWallet a = factory.createAccount{value: 1e18}(owners, 0);
+        vm.expectCall(address(a), abi.encodeCall(CoinbaseSmartWallet.initialize, (owners)), 0);
+        CoinbaseSmartWallet a2 = factory.createAccount(owners, 0);
+        assertEq(address(a), address(a2));
+    }
+
+    function testRevertsIfLength32ButLargerThanAddress() public {
```
Comment (wilsoncusack) [inline test/CoinbaseSmartWalletFactory.t.sol:43]: our test function names are not consistent right now, but let's at least try to pick one format when adding new ones

File: test/CoinbaseSmartWalletFactory.t.sol (hunk 2 above)
Comment (stevieraykatz) [reply test/CoinbaseSmartWalletFactory.t.sol:36]: So the `test_exitIfAccountIsAlreadyInitialized` was intended to catch the branch that foundry coverage was missing. In messing with MultiOwnable, I noticed that this test ticked the coverage for Factory. I think that the difference between the impl. here and the one in MultiOwnable is that the actual factory is included.

**Review by wilsoncusack — APPROVED** [review]

**Pull request merged**